aperturedb/BBoxDataCSV.py

```diff
@@ -69,35 +69,36 @@ def __init__(self, filename):
         self.img_key = self.header[0]
         self.command = "AddBoundingBox"
 
-    def getitem(self, idx):
-
-        val = self.df.loc[idx, self.img_key]
-        box_data_headers = [HEADER_X_POS,
-                            HEADER_Y_POS, HEADER_WIDTH, HEADER_HEIGHT]
-        box_data = [int(self.df.loc[idx, h]) for h in box_data_headers]
+    def getitem(self, idx, ctx):
 
         q = []
 
-        ref_counter = idx + 1
-        # TODO we could reuse image references within the batch
-        # instead of creating a new find for every image.
-        img_ref = ref_counter
-        fi = {
-            "FindImage": {
-                "_ref": img_ref,
+        if "img_id_to_ref" not in ctx:
+            ctx["img_id_to_ref"] = {}
+        img_map = ctx["img_id_to_ref"]
+
+        img_id = self.df.loc[idx, self.img_key]
+        if img_id not in img_map:
+            img_ref = len(img_map) + 1
+            img_map[img_id] = img_ref
+
+            fi = {
+                "FindImage": {
+                    "_ref": img_ref,
+                    "constraints": {
+                        self.img_key: ["==", img_id],
+                    },
+                },
             }
-        }
+            q.append(fi)
 
-        key = self.img_key
-        val = val
-        constraints = {}
-        constraints[key] = ["==", val]
-        fi["FindImage"]["constraints"] = constraints
-        q.append(fi)
+        box_data_headers = [HEADER_X_POS,
+                            HEADER_Y_POS, HEADER_WIDTH, HEADER_HEIGHT]
+        box_data = [int(self.df.loc[idx, h]) for h in box_data_headers]
 
         rect_attrs = ["x", "y", "width", "height"]
         custom_fields = {
-            "image": img_ref,
+            "image_ref": img_map[img_id],
```

Comment:
NOTE: this is a breaking change!  it will only work with https://github.com/aperture-data/athena/pull/518